Zipkin dependency job should support cassandra configured with ssl

Zipkin dependency job does not support cassandra configured with ssl. Job throws 'channel closed' error. 

Zipkin service supports cassandra over ssl with property CASSANDRA_USE_SSL. Also while running jar, we need to pass java arguments - javax.net.ssl.trustStore, javax.net.ssl.trustStorePassword, javax.net.ssl.keyStore, javax.net.ssl.keyStorePassword

Error stacktrace -
`Exception in thread "main" java.io.IOException: Failed to open native connection to Cassandra at {10.100.100.102, 10.100.100.101}:9042
        at com.datastax.spark.connector.cql.CassandraConnector$.com$datastax$spark$connector$cql$CassandraConnector$$createSession(CassandraConnector.scala:162)
        at com.datastax.spark.connector.cql.CassandraConnector$$anonfun$3.apply(CassandraConnector.scala:148)
        at com.datastax.spark.connector.cql.CassandraConnector$$anonfun$3.apply(CassandraConnector.scala:148)
        at com.datastax.spark.connector.cql.RefCountedCache.createNewValueAndKeys(RefCountedCache.scala:31)
        at com.datastax.spark.connector.cql.RefCountedCache.acquire(RefCountedCache.scala:56)
        at com.datastax.spark.connector.cql.CassandraConnector.openSession(CassandraConnector.scala:81)
        at com.datastax.spark.connector.cql.CassandraConnector.withSessionDo(CassandraConnector.scala:109)
        at com.datastax.spark.connector.cql.CassandraConnector.withClusterDo(CassandraConnector.scala:120)
        at com.datastax.spark.connector.cql.Schema$.fromCassandra(Schema.scala:322)
        at com.datastax.spark.connector.cql.Schema$.tableFromCassandra(Schema.scala:342)
        at com.datastax.spark.connector.rdd.CassandraTableRowReaderProvider$class.tableDef(CassandraTableRowReaderProvider.scala:50)
        at com.datastax.spark.connector.rdd.CassandraTableScanRDD.tableDef$lzycompute(CassandraTableScanRDD.scala:61)
        at com.datastax.spark.connector.rdd.CassandraTableScanRDD.tableDef(CassandraTableScanRDD.scala:61)
        at com.datastax.spark.connector.rdd.CassandraTableRowReaderProvider$class.verify(CassandraTableRowReaderProvider.scala:137)
        at com.datastax.spark.connector.rdd.CassandraTableScanRDD.verify(CassandraTableScanRDD.scala:61)
        at com.datastax.spark.connector.rdd.CassandraTableScanRDD.getPartitions(CassandraTableScanRDD.scala:231)
        at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:239)
        at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:237)
        at scala.Option.getOrElse(Option.scala:120)
        at org.apache.spark.rdd.RDD.partitions(RDD.scala:237)
        at com.datastax.spark.connector.rdd.SpannedRDD.getPartitions(SpannedRDD.scala:25)
        at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:239)
        at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:237)
        at scala.Option.getOrElse(Option.scala:120)
        at org.apache.spark.rdd.RDD.partitions(RDD.scala:237)
        at org.apache.spark.rdd.MapPartitionsRDD.getPartitions(MapPartitionsRDD.scala:35)
        at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:239)
        at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:237)
        at scala.Option.getOrElse(Option.scala:120)
        at org.apache.spark.rdd.RDD.partitions(RDD.scala:237)
        at org.apache.spark.rdd.MapPartitionsRDD.getPartitions(MapPartitionsRDD.scala:35)
        at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:239)
        at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:237)
        at scala.Option.getOrElse(Option.scala:120)
        at org.apache.spark.rdd.RDD.partitions(RDD.scala:237)
        at org.apache.spark.rdd.MapPartitionsRDD.getPartitions(MapPartitionsRDD.scala:35)
        at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:239)
        at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:237)
        at scala.Option.getOrElse(Option.scala:120)
        at org.apache.spark.rdd.RDD.partitions(RDD.scala:237)
        at org.apache.spark.Partitioner$.defaultPartitioner(Partitioner.scala:65)
        at org.apache.spark.api.java.JavaPairRDD.reduceByKey(JavaPairRDD.scala:526)
        at zipkin.dependencies.cassandra.CassandraDependenciesJob.run(CassandraDependenciesJob.java:143)
        at zipkin.dependencies.ZipkinDependenciesJob.main(ZipkinDependenciesJob.java:38)
Caused by: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /10.100.100.101:9042 (com.datastax.driver.core.exceptions.TransportException: [/10.100.100.101] Channel has been closed), /10.100.100.102:9042 (com.datastax.driver.core.exceptions.TransportException: [/10.100.100.102] Channel has been closed))
        at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:231)
        at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:77)
        at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:1414)
        at com.datastax.driver.core.Cluster.getMetadata(Cluster.java:393)
        at com.datastax.spark.connector.cql.CassandraConnector$.com$datastax$spark$connector$cql$CassandraConnector$$createSession(CassandraConnector.scala:155)
`

--- a/cassandra/src/main/java/zipkin/dependencies/cassandra/CassandraDependenciesJob.java
+++ b/cassandra/src/main/java/zipkin/dependencies/cassandra/CassandraDependenciesJob.java
@@ -56,13 +56,16 @@ public final class CassandraDependenciesJob {
     /** Comma separated list of hosts / IPs part of Cassandra cluster. Defaults to localhost */
     String contactPoints = getEnv("CASSANDRA_CONTACT_POINTS", "localhost");
 
-    Map<String, String> sparkProperties = ImmutableMap.of(
-        "spark.ui.enabled", "false",
-        "spark.cassandra.connection.host", parseHosts(contactPoints),
-        "spark.cassandra.connection.port", parsePort(contactPoints),
-        "spark.cassandra.auth.username", getEnv("CASSANDRA_USERNAME", ""),
-        "spark.cassandra.auth.password", getEnv("CASSANDRA_PASSWORD", "")
-    );
+    Map<String,String> sparkProperties = ImmutableMap.<String, String>builder()
+      .put("spark.ui.enabled", "false")
+      .put("spark.cassandra.connection.host", parseHosts(contactPoints))
+      .put("spark.cassandra.connection.port", parsePort(contactPoints))
+      .put("spark.cassandra.connection.ssl.enabled", getEnv("CASSANDRA_USE_SSL", "false"))
+      .put("spark.cassandra.connection.ssl.trustStore.password", getEnv("CASSANDRA_KEYSTORE_PASSWORD", ""))
+      .put("spark.cassandra.connection.ssl.trustStore.path", getEnv("CASSANDRA_KEYSTORE_PATH", ""))
+      .put("spark.cassandra.auth.username", getEnv("CASSANDRA_USERNAME", ""))
+      .put("spark.cassandra.auth.password", getEnv("CASSANDRA_PASSWORD", ""))
+      .build();
 
     // local[*] master lets us run & test the job locally without setting a Spark cluster
     String sparkMaster = getEnv("SPARK_MASTER", "local[*]");

